chainforge/react-server/src/backend/query.ts

```diff
@@ -291,8 +300,10 @@ export class PromptPipeline {
     let query: Dict | undefined;
     let response: Dict | LLMResponseError;
     try {
+      if (check_stop_condition && check_stop_condition())  { throw new UserForcedPrematureExit(); }
       [query, response] = await call_llm(llm, prompt.toString(), n, temperature, params);
```

Comment:
You should also perform this check directly after `call_llm` here... I know it looks weird but, basically, anytime we "await" we are waiting some time for a response to return. We can't do anything about canceling the call_llm itself, but, we should check immediately after it returns a value, whether to cancel.